src/vos/vos_common.c

```diff
@@ -776,57 +780,70 @@ vos_self_init(const char *db_path, bool use_sys_db, int tgt_id)
 {
 	char		*evt_mode;
 	int		 rc = 0;
-	char		*lmm_db_path = getenv("DAOS_LMM_DB_PATH");
+	char		*lmm_db_path = NULL;
 	struct sys_db	*db;
 
 	D_MUTEX_LOCK(&self_mode.self_lock);
 	if (self_mode.self_ref) {
 		self_mode.self_ref++;
-		D_GOTO(out, rc);
+		goto out;
 	}
 
 	rc = ABT_init(0, NULL);
 	if (rc != 0)
-		D_GOTO(out, rc);
+		goto out;
 
 	vos_start_epoch = 0;
 
 #if VOS_STANDALONE
 	self_mode.self_tls = vos_tls_init(0, -1);
 	if (!self_mode.self_tls) {
 		ABT_finalize();
-		D_GOTO(out, rc);
+		goto out;
 	}
 #endif
 	rc = vos_mod_init();
 	if (rc)
-		D_GOTO(failed, rc);
+		goto failed;
+
+	if (bio_nvme_configured(SMD_DEV_TYPE_META)) {
+		bool	needs_free = false;
+
+		rc = bio_get_lmm_db_path(lmm_db_path);
```

Comment:
Use the passed in 'db_path'?

Reply:
and that is because in vos_self_init() there won't be an external Meta data storage path? i.e. LMM DB can live in dss_storage_path? is that why?

Reply:
Right, to minimize code changes, I think we could store the lmm DB in the storage path for VOS unit tests and vos_perf.

Reply:
done